lib/couchdb_config.py

```diff
@@ -0,0 +1,169 @@
+        required: true
+    value:
+        description:
+            - The configuration value
+        required: true
+    host:
+        description:
+            - The host running the database
+        required: false
+        default: localhost
+    port:
+        description:
+            - The port to connect to
+        required: false
+        default: 5984
+    user:
+        description:
+            - The administrator user used to authenticate with
+        required: false
+    password:
+        description:
+            - The administrator password of the user
+        required: false
+'''
+
+EXAMPLES = '''
+# setting multiple values
+- couchdb_config: >
+    section={{ item.section }}
+    key={{ item.key }}
+    value={{ item.value }}
+  with_items:
+    - { section: 'couch_httpd_auth', key: 'timeout', value: '60' }
+    - { section: 'socket_keys', key: 'worker_batch_size', value: '510' }
+
+# setting a list of tuples
+- couchdb_config: >
+    section='httpd'
+    key='socket_keys'
+    value='[{recbuf, 262140}, {sndbuf, 262140}]'
+
+# setting a value when CouchDB is secured
+- couchdb_config: >
+    user=admina
+    password=@dm1na
+    section='couch_httpd_auth'
+    key='timeout'
+    value='60'
+
+# changing host and port
+- couchdb_config: >
+    host=192.168.33.100
+    port=5985
+    section='couch_httpd_auth'
+    key='timeout'
+    value='60'
+'''
+
+IS_INSTALLED_PYCOUCHDB = True
+try:
+    import pycouchdb
+    from pycouchdb.resource import Resource
+except ImportError:
+    IS_INSTALLED_PYCOUCHDB = False
+
+IS_INSTALLED_REQUESTS = True
+try:
+    import requests
+    from requests.exceptions import ConnectionError
+except ImportError:
+    IS_INSTALLED_REQUESTS = False
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec=dict(
+            host=dict(default="localhost"),
+            port=dict(type='int', default=5984),
+            user=dict(type='str'),
+            password=dict(type='str', no_log=True),
+            section=dict(type='str', required=True),
+            key=dict(type='str', required=True),
+            value=dict(type='str', required=True),
+        )
+    )
+
+    if not IS_INSTALLED_PYCOUCHDB:
+        module.fail_json(msg="Please install pycouchdb")
+    if not IS_INSTALLED_REQUESTS:
+        module.fail_json(msg="Please install requests")
+
+    couchdb_uri = create_couchdb_uri(module.params["host"],
+                                     module.params["port"],
+                                     user=module.params["user"],
+                                     password=module.params["password"])
+
+    try:
+        pycouchdb.Server(couchdb_uri).info()
+    except ConnectionError:
+        module.fail_json(msg='Failed to connect to ' + couchdb_uri)
+
+    section = module.params["section"]
+    key = module.params["key"]
+    value = module.params["value"]
+
+    response = Resource(couchdb_uri).put("_config/{0}/{1}".format(section, key), data="\"{0}\"".format(value))
```

Comment:
This can raise exceptions, which should be caught and converted into a proper `module.fail_json` response.

From the [developer guide](http://docs.ansible.com/ansible/developing_modules.html#module-checklist):

> Exceptions: The module must handle them. (exceptions are bugs)